test/valid-data/enums-union/schema.json

```diff
@@ -0,0 +1,141 @@
+{
+  "$ref": "#/definitions/MyObject",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "definitions": {
+    "MyObject": {
+      "additionalProperties": false,
+      "properties": {
+        "enumMemberWithInterface": {
+          "anyOf": [
+            {
+              "const": "alpha",
+              "type": "string"
+            },
+            {
+              "additionalProperties": false,
+              "properties": {
+                "abc": {
+                  "type": "string"
+                }
+              },
+              "required": [
+                "abc"
+              ],
+              "type": "object"
+            }
+          ]
+        },
+        "enumMemberWithLiteral": {
+          "enum": [
+            "alpha",
+            "foo"
+          ],
+          "type": "string"
+        },
+        "enumMemberWithLiteralAndNull": {
+          "enum": [
+            "alpha",
+            "foo",
+            null
+          ],
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "enumMembers": {
+          "enum": [
+            "alpha",
+            "beta"
+          ],
+          "type": "string"
+        },
+        "enumMembersWithNumber": {
+          "enum": [
+            "alpha",
+            666
+          ],
+          "type": [
+            "string",
+            "number"
+          ]
+        },
+        "twoDifferentEnumMembers": {
+          "enum": [
+            "alpha",
+            2
+          ],
+          "type": [
+            "string",
+            "number"
+          ]
+        },
+        "twoDifferentWholeEnums": {
+          "enum": [
```

Comment:
This is a union of two enums, this type should be a anyOf 2 references so we can reuse json schemas.

Reply:
i think If you export those enums, this will become a reference

Reply:
https://github.com/vega/ts-json-schema-generator/pull/2149#issuecomment-2576020653